app/uk/gov/hmrc/mobilepaye/domain/taxcalc/P800Summary.scala

```diff
@@ -26,45 +27,41 @@ import uk.gov.hmrc.time.TaxYear
 import scala.util.{Failure, Success, Try}
 
 case class P800Summary(
-                        p800_status:   P800Status,
-                        paymentStatus: RepaymentStatus,
-                        amount:        BigDecimal,
-                        taxYear:       Int,
-                        datePaid:      Option[LocalDate]
-                      )
+  _type:    P800Status,
+  status:   RepaymentStatus,
+  amount:   BigDecimal,
+  datePaid: Option[LocalDate]
+)
 
 object P800Summary {
-  def toP800Repayment(p800Summary: P800Summary): Option[P800Repayment] = {
-    def withOverpaidP800(p800Summary: P800Summary): Option[P800Summary] = {
-      p800Summary.p800_status match {
+  def toP800Repayment(p800Summary: P800Summary, taxYear: Int): Option[P800Repayment] = {
+    def withOverpaidP800(p800Summary: P800Summary): Option[P800Summary] =
+      p800Summary._type match {
         case Overpaid => Option(p800Summary)
         case _        => None
       }
-    }
 
-    def withAcceptableRepaymentStatus(p800Summary: P800Summary): Option[P800Summary] = {
-      p800Summary.paymentStatus match {
-          case SaUser        => None
-          case UnableToClaim => None
-          case _             => Option(p800Summary)
-        }
-    }
+    def withAcceptableRepaymentStatus(p800Summary: P800Summary): Option[P800Summary] =
+      p800Summary.status match {
+        case SaUser        => None
+        case UnableToClaim => None
+        case _             => Option(p800Summary)
+      }
 
     def transform(p800Summary: P800Summary): P800Repayment = {
-      def withLink: Option[String] = {
-        p800Summary.paymentStatus match {
-          case Refund => Option(s"/tax-you-paid/${TaxYear.current.currentYear - 1}-${TaxYear.current.currentYear}/paid-too-much")
+      def withLink: Option[String] =
+        p800Summary.status match {
+          case Refund => Option(s"/tax-you-paid/${taxYear - 1}-${TaxYear.current.currentYear}/paid-too-much")
```

Comment:
`TaxYear.current.currentYear` should be `TaxYear(yaxYear)` otherwise it could be be 2016-2019 if the url has 2016 in it